src/substrate/substrate.controller.ts

```diff
@@ -67,13 +67,14 @@ export class SubstrateController {
       const bindingEth = await this.substrateService.bindEthAddressToSubstrateAddress(
         ethAddress,
         accountId,
-      );
-      
+      );      
       if (bindingEth) {
-        substrateAddress = await this.substrateService.getSubstrateAddressByEthAddress(ethAddress);
-        await this.substrateService.sendReward(accountId, rewardAmount);
-        reward = rewardAmount
-        await this.rewardService.insert(dataInput);
+        const isRewardHasBeenSend = await this.rewardService.getRewardBindingByAccountId(accountId)
+        if(!isRewardHasBeenSend){
+          await this.substrateService.sendReward(accountId, rewardAmount);
+          reward = rewardAmount
+          await this.rewardService.insert(dataInput);
+        }
       } else {
         response.status(401).send('Binding Error');
       }
```

Comment:
Too many `if`s refactor this logic.

Reply:
because to much condition for this wallet binding flow, any idea for change the flow wallet binding?

Reply:
How about this:
```
let substrateAddress =
    await this.substrateService.getSubstrateAddressByEthAddress(ethAddress);

// ngecek eth address sudah pernah binding blm
if (substrateAddress) {
    // kalau udah dy lgsung binding tanpa reward
    await this.substrateService.bindEthAddressToSubstrateAddress(
        ethAddress,
        accountId,
    );
    
    return response
        .status(200)
        .send({
        reward,
        message :`eth-address ${ethAddress} bound to ${accountId}`
        });
}

const bindingEth = await this.substrateService.bindEthAddressToSubstrateAddress(
    ethAddress,
    accountId,
);      

// kalau belum dy binding, trus cek lagi udah ke binding apa g
if (!bindingEth) {
    // kalau g juga kebinding berarti kirim 401 binding error
    return response.status(401).send('Binding Error');
}

// kalau udah dy cek lagi di bawah
const isRewardHasBeenSend = await this.rewardService.getRewardBindingByAccountId(accountId)

//  acount id nya udah pernah di kirim reward yg wallet binding apa g
if(!isRewardHasBeenSend){
    // kalau blm dy d kasih reward
    await this.substrateService.sendReward(accountId, rewardAmount);
    reward = rewardAmount
    await this.rewardService.insert(dataInput);
}

return response
    .status(200)
    .send({
    reward,
    message :`eth-address ${ethAddress} bound to ${accountId}`
    });
```
I made it according to your comments on Discord. Would you mind removing the comments when done?